libs/utils/include/utils/algorithm.h

```diff
@@ -203,6 +203,19 @@ bit_cast(const From &src) noexcept {
     return reinterpret_cast<const To&>(src);
 }
 
+template<typename T, typename Pred>
+T const* find_range(T const* const begin, T const* end, Pred pred) noexcept {
```

Comment:
The name `find_range` is confusing to me, since this returns a single item. Maybe `find_first_nonequal`?

Also, I think you can make `end` also be `T const* const end`.

Reply:
very good point. In fact we can use `std::find_if_not` instead. thanks!